business_requirement_deliverable_sale/models/business.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2019 Tecnativa Victor M.M. Torres>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class BusinessRequirement(models.Model):
+    _inherit = 'business.requirement'
+
+    sale_order_ids = fields.One2many(
+        comodel_name='sale.order',
+        inverse_name='business_requirement_id',
+        string='Orders',
+    )
+    order_count = fields.Integer(
+        string='Count',
+        compute='_compute_order_count',
+    )
+    detailed_order = fields.Boolean(
+        help='Check this if need generated a detailed order'
+    )
+
+    @api.multi
+    @api.depends('sale_order_ids')
+    def _compute_order_count(self):
+        for order in self:
+            order.order_count = len(order.sale_order_ids)
+
+    @api.multi
+    def convert_requirement_sale(self):
+        vals = {
+            'br_id': self.id,
+            'brd_ids': [(6, 0, self.deliverable_lines.ids)]
+        }
+        wizard_obj = self.env['convert.requirement.sale']
+        wizard = wizard_obj.create(vals)
+        action = wizard.wizard_view()
+        return action
+
+    @api.multi
+    def launch_convert_wizard(self):
+        vals = {
+            'br_id': self.id,
+            'brd_ids': [(6, 0, self.deliverable_lines.ids)],
+        }
+        wizard_obj = self.env['convert.requirement.sale']
+        wizard = wizard_obj.create(vals)
+        action = wizard.wizard_view()
+        return action
+
+    @api.multi
+    def open_orders(self):
```

Comment:
See this example
https://github.com/OCA/stock-logistics-workflow/blob/d87ee07a99f8d90ec74713e2c7e83aa6e7f55b90/stock_batch_picking/models/stock_batch_picking.py#L212